plugins/filter/ipaddr.py

```diff
@@ -0,0 +1,281 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+filter plugin file for ipaddr filters: cidr_merge
+"""
+from __future__ import absolute_import, division, print_function
+from functools import partial
+from ansible_collections.ansible.utils.plugins.plugin_utils.base.ipaddr_utils import (
+    ipaddr,
+    _need_netaddr,
+)
+from ansible.errors import AnsibleFilterError
+from ansible_collections.ansible.utils.plugins.module_utils.common.argspec_validate import (
+    AnsibleArgSpecValidator,
+)
+
+__metaclass__ = type
+
+
+try:
+    from jinja2.filters import pass_environment
+except ImportError:
+    from jinja2.filters import environmentfilter as pass_environment
+
+try:
+    import netaddr
+
+    HAS_NETADDR = True
+except ImportError:
+    # in this case, we'll make the filters return error messages (see bottom)
+    HAS_NETADDR = False
+else:
+
+    class mac_linux(netaddr.mac_unix):
+        pass
+
+    mac_linux.word_fmt = "%.2x"
+
+DOCUMENTATION = """
+    name: ipaddr
+    author: Ashwini Mhatre (@amhatre)
+    version_added: "2.5.0"
+    short_description: This filter is designed to return the input value if a query is True, else False.
+    description:
+        - This filter is designed to return the input value if a query is True, and False if a query is False
+        - This way it can be easily used in chained filters
+    options:
+        value:
+            description:
+            - list of subnets or individual address or any other values input for ipaddr plugin
+            type: list
+            elements: str
+            required: True
+        query:
+            description:
+            - You can provide a single argument to each ipaddr() filter.
+            - The filter will then treat it as a query and return values modified by that query.
+            - Types of queries include:
+                - query by name: ansible.utils.ipaddr('address'), ansible.utils.ipv4('network');
+                - query by CIDR range: ansible.utils.ipaddr('192.168.0.0/24'), ansible.utils.ipv6('2001:db8::/32');
+                - query by index number: ansible.utils.ipaddr('1'), ansible.utils.ipaddr('-1');
```

Comment:
I realize https://github.com/ansible-collections/ansible.utils/blob/main/plugins/plugin_utils/base/ipaddr_utils.py#L396 is a huge list, but we should document what you can pass in to ipaddr somewhere.

Reply:
Created one issue for this https://github.com/ansible-collections/ansible.utils/issues/130